### DISCUSSION TITLE
Type Booster

### DISCUSSION THREAD

**Discussion by marcorocca1973** [DevTrevi/DgcReader#33] [General]
Ciao
ho visto che è stata rilasciata per la C19 e la nuova tipologia "Booster" come tipo di verifica. Hai già in piano una nuova versione 2.1.0 per gestire anche questa casistica (nel caso ti servisse supporto per qualche "update" a disposizione :) )
Grz per il tuo prezioso lavoro!
Marco

**Comment by DevTrevi** [comment]
Ciao @marcorocca1973 , sì ho già pianificato un rilascio per allineare la libreria alla versione 1.1.1 dell'sdk, entro fine settimana dovrei riuscire a realizzarlo

**Reply by DevTrevi** [reply]
Ciao @marcorocca1973 , ho rilasciato la versione 2.1.0 che include la modalità booster #40

**Reply by marcorocca1973** [reply]
Ciao @DevTrevi ho visto.... Ho costruito una Azure Function che la richiama in modo da poter creare un custom connector per la Power Platform. Funziona perfettamente!

**Reply by DevTrevi** [reply]
Ottimo! 😉🚀

**Reply by marcorocca1973** [reply]
Ho solo un problema con ILogger.
Nella mia AzureFn ho il metodo

public static async Task<HttpResponseMessage> Run(
            [HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req,
            ILogger log, ExecutionContext context)
        {

per cui tutto quanto viene loggato in questo metodo con statement del tipo

> log.LogInformation("DgcReader trigger function processed a request");

viene correttamente visualizzato.
Quando creo il servizio

services.AddDgcReader(
                )                     // Add the DgcReaderService as singleton
                .AddItalianTrustListProvider(o =>
                {
                    o.BasePath = context.FunctionAppDirectory;
               
                })
                .AddItalianRulesValidator(o =>
                {
                    o.ValidationMode = vType;
                    o.BasePath = context.FunctionAppDirectory;
                    

                })
                ;

ho provato ad utilizzare AddLogging ma non viene recepito, per cui tutto quanto viene loggato nei metodi "interni" non viene visualizzato. 

Non riesco a capire come "passare" il mio ILogging all'oggetto ILogger<DgcReaderService> :/
Ti viene in mente dove possa sbagliare?

**Reply by DevTrevi** [reply]
Ciao @marcorocca1973 , se utilizzi la DI come vedo, il servizio `ILogger` viene già passato ai servizi DgcReader registrati.
Il problema nel tuo caso dovrebbe essere solo il fatto che nelle Azure Functions le categorie aggiuntive di logging sono filtrate di default.

Dovrebbe essere sufficiente configurare il log level per DgcReader nel file `host.json`, come descritto nella documentazione: [Use dependency injection in .NET Azure Functions](https://docs.microsoft.com/en-us/azure/azure-functions/functions-dotnet-dependency-injection#iloggert-and-iloggerfactory)

Puoi aggiungere il namespace principale `DgcReader`, o se preferisci i singoli servizi (`DgcReader.DgcReaderService`, `DgcReader.BlacklistProviders.Italy.ItalianDrlBlacklistProvider` ...)

Esempio:
``` json
{
    "version": "2.0",
    "logging": {
        "applicationInsights": {
            "samplingSettings": {
                "isEnabled": true,
                "excludedTypes": "Request"
            }
        },
        "logLevel": {
            "DgcReader": "Information",
            "DgcReader.BlacklistProviders.Italy.ItalianDrlBlacklistProvider": "Debug",
        }
    }
}
```

**Reply by marcorocca1973** [reply]
Ciao
ho risolto usando il "vecchio" metodo Create :)

var lgtrustListProvider = _loggerFactory.CreateLogger<ItalianTrustListProvider>();
            var trustListProvider = ItalianTrustListProvider.Create(httpClient, new ItalianTrustListProviderOptions
            {
                BasePath = context.FunctionAppDirectory
            }, lgtrustListProvider);

            var lgblacklistProvider = _loggerFactory.CreateLogger<ItalianDrlBlacklistProvider>();
            var blacklistProvider = ItalianDrlBlacklistProvider.Create(httpClient, new
                ItalianDrlBlacklistProviderOptions
                {
                    BasePath = context.FunctionAppDirectory
                }, lgblacklistProvider);

**Reply by DevTrevi** [reply]
Ottimo, se ti è possibile ti consiglierei comunque di fare un tentativo col metodo precedente utilizzando la DI e la configurazione di logLevel. 
Altrimenti se non lo hai gia fatto, ricorda di registrare entrambi i blacklist provider (Drl e RulesValidator) come blacklist provider, altrimenti avrai una validazione parziale delle blacklist. 
Nei prossimi giorni aggiornerò il readme per rendere più chiaro questo passaggio
